crates/core/tests/client.rs

```diff
@@ -1,22 +1,137 @@
+mod helpers;
+
 #[cfg(test)]
 mod tests {
 
     use std::str::FromStr;
 
+    use dojo_test_utils::sequencer::TestSequencer;
+    use ethers::types::{Address as EthersAddress, U256 as EthersU256};
     use kakarot_rpc_core::client::client_api::{KakarotEthApi, KakarotStarknetApi};
+    use kakarot_rpc_core::client::config::StarknetConfig;
     use kakarot_rpc_core::client::errors::EthApiError;
+    use kakarot_rpc_core::client::KakarotClient;
     use kakarot_rpc_core::mock::wiremock_utils::setup_mock_client_crate;
     use kakarot_rpc_core::models::block::BlockWithTxs;
     use kakarot_rpc_core::models::convertible::{ConvertibleStarknetBlock, ConvertibleStarknetEvent};
     use kakarot_rpc_core::models::event::StarknetEvent;
-    use kakarot_rpc_core::models::felt::Felt252WrapperError;
+    use kakarot_rpc_core::models::felt::{Felt252Wrapper, Felt252WrapperError};
     use kakarot_rpc_core::models::ConversionError;
     use reth_primitives::{Address, Bytes, H256};
     use reth_rpc_types::Log;
     use starknet::core::types::{BlockId, BlockTag, Event, FieldElement};
     use starknet::core::utils::get_selector_from_name;
     use starknet::providers::Provider;
 
+    use crate::helpers::constants::{EVM_ADDRESS, EVM_PRIVATE_KEY};
+    use crate::helpers::deploy_helpers::{create_raw_ethereum_tx, deploy_kakarot_system};
+
+    #[tokio::test]
+    #[ignore]
+    async fn test_constructable() {
+        // initial setup of Constructable to test we can deploy contracts w/ constructor arguments
+        let starknet_test_sequencer = TestSequencer::start().await;
+
+        let eoa_private = H256::from_slice(&hex::decode(EVM_PRIVATE_KEY).unwrap());
+
+        let expected_funded_amount = FieldElement::from_dec_str("100000000000000000").unwrap();
+
+        let deployed_kakarot =
+            deploy_kakarot_system(&starknet_test_sequencer, EVM_ADDRESS, eoa_private, expected_funded_amount).await;
+
+        let address = "0x0000000000000000000000000000000000000000".parse::<EthersAddress>().unwrap();
+        let (_constructable_abi, deployed_addresses) = deployed_kakarot
```

Comment:
why do we deploy using kakarot directly and not using evm things?

Reply:
not sure about this question but: 
because we don't have access to an ethereum json rpc spec -> we cannot use ether-rs 

we only have access to the kakarot client, it does not implement json rpc of eth

Reply:
yeah, that's the crux of it. we are 'inbetween' starknet and ethereum at this stage. 

we would want to test at the rpc level the rpc's ability to do deployment, i agree.